v3/src/models/formula/functions/string-functions.ts

```diff
@@ -0,0 +1,210 @@
+    }
+  },
+  // includes(stringToLookIn, stringToFind) Takes two arguments and returns true if the second argument
+  // is a substring of the first (also treated as a string).
+  includes: {
+    numOfRequiredArguments: 2,
+    evaluate: (...args: FValue[]) => {
+      const stringToLookIn = valueToString(args[0]),
+        stringToFind = valueToString(args[1])
+      return stringToLookIn.includes(stringToFind)
+    }
+  },
+  // join(delimiter, string1, string2, …) Returns the string formed by concatenating its string arguments, each
+  // seperated by its delimiter argument.
+  join: {
+    numOfRequiredArguments: 2,
+    evaluate: (...args: FValue[]) => {
+      const delimiter = valueToString(args[0])
+      return args.slice(1).map(valueToString).join(delimiter)
+    }
+  },
+  // patternMatches(stringToLookIn, pattern) Returns the number of times in string_to_look_in the string or
+  // regular expression represented by pattern occurs. The search is case-insensitive.
+  patternMatches: {
+    numOfRequiredArguments: 2,
+    evaluate: (...args: FValue[]) => {
+      const stringToLookIn = valueToString(args[0]),
+        pattern = valueToString(args[1])
+      return (stringToLookIn.match(new RegExp(pattern, "gmi")) || []).length
+    }
+  },
+  // repeatString(aString, numRepetitions) Takes two arguments, the first a string and the second an integer ≥ 0.
+  // Returns the result of concatenating aString numRepetitions times.
+  repeatString: {
+    numOfRequiredArguments: 2,
+    evaluate: (...args: FValue[]) => {
+      const aString = valueToString(args[0]),
+        numRepetitions = Number(args[1])
+      return aString.repeat(numRepetitions)
+    }
+  },
+  // replaceChars(aString, start, numChars, substituteString) Takes four arguments. The first is the original string.
+  // The second is an integer > 0 specifying the starting location for the substitution. The third is the number of
+  // characters to be replaced, and the last is the string that is to be substituted.
+  // If numChars is 0, substituteString is inserted.
+  replaceChars: {
+    numOfRequiredArguments: 4,
+    evaluate: (...args: FValue[]) => {
+      const aString = valueToString(args[0]),
+        start = Number(args[1]),
+        numChars = Number(args[2]),
+        substituteString = valueToString(args[3])
+      return aString.substring(0, start - 1) + substituteString + aString.substring(start + numChars - 1)
+    }
+  },
+  // replaceString(aString, stringToFind, substituteString) Takes three string arguments and substitutes the third
+  // for all occurrences of the second in the first.
+  replaceString: {
+    numOfRequiredArguments: 3,
+    evaluate: (...args: FValue[]) => {
+      const aString = valueToString(args[0]),
+        stringToFind = valueToString(args[1]),
+        substituteString = valueToString(args[2])
+      return aString.replace(new RegExp(stringToFind, "g"), substituteString)
```

Comment:
Strings, particularly user-entered strings, should always be escaped before using them as regular expressions.
```suggestion
      return aString.replace(new RegExp(escapeStringRegexp(stringToFind), "g"), substituteString)
```